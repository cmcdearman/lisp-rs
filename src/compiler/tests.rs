use crate::{
    parser::ast::{Expr, Lit},
    span::{Span, Spanned},
};

#[test]
fn test_compile_int() {
    let expr = (Expr::Lit(Lit::Int(123)), Span::from(0..0));
    let mut c = crate::compiler::Compiler::new();
    let chunk = c.compile(&expr).expect("Failed to compile");
    insta::assert_debug_snapshot!(chunk);
}
