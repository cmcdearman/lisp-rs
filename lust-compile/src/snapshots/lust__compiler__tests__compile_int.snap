---
source: src/compiler/tests.rs
expression: chunk
---
0000 <0, 0> CONST    0 '123'

